tests/test_local_env.py

```diff
@@ -117,3 +117,12 @@ def test_resolve_dependencies__in_fake_venv__returns_local_and_id_deps(fake_venv
         "pandas": Package.identity_mapping("pandas"),
         "empty-pkg": Package("empty_pkg", {DependenciesMapping.LOCAL_ENV: set()}),
     }
+
+
+def test_on_pulled_venv__returns_local_deps():
+    actual = resolve_dependencies(["pandas", "click"], pyenv_path=None, pull_deps=True)
+    assert actual == {
+        "pandas": Package("pandas", {DependenciesMapping.LOCAL_ENV: {"pandas"}}),
+        "click": Package("click", {DependenciesMapping.LOCAL_ENV: {"click"}}),
+    }
+    assert actual["pandas"].is_used(["pandas"])
```

Comment:
I'd like to avoid hitting the network in unit tests if we can. Can we configure/customize the `pip install` calls to use a local directory instead? Looking at `pip install --help`, there is an `--index-url` option that we might be able to use:

> Base URL of the Python Package Index (default https://pypi.org/simple). This should point to a repository compliant with PEP 503 (the simple repository API) or a local directory laid out in the same format.

Googling a little bit suggests that the same can be achieved by setting the `PIP_INDEX_URL` environment variable, so we can even customize this in tests _without_ changing the `pip install` command line.

That said, I'm happy to have _integration_ tests talk directly to PyPI to verify that this part works as well.

Reply:
Sat up (yet another) context manager in unit tests to run this particular test with suitable env variables to point towards local PyPI packages. See [836c4e1](https://github.com/tweag/FawltyDeps/pull/252/commits/836c4e1fd94f02391e96fd02b94bab27ac180fa5).